File: 190.reverse-bits.js
/*
 * @lc app=leetcode id=190 lang=javascript
 *
 * [190] Reverse Bits
 */

// @lc code=start
/**
 * @param {number} n - a positive integer
 * @return {number} - a positive integer
 */
var reverseBits = function(n) {
  let binary = Number(n).toString(2)
  binary = '0'.repeat(32-binary.length)+binary
  return parseInt([...binary].reverse().join(''),2)
};
// @lc code=end

File: 190.reverse-bits.py
#
# @lc app=leetcode id=190 lang=python3
#
# [190] Reverse Bits
#

# @lc code=start
class Solution:
    def reverseBits(self, n: int) -> int:
        return int(bin(n)[:1:-1].ljust(32,'0'),2)
# @lc code=end